### PR TITLE
[feat] SampleGen: Parse empty object literals

### PR DESCRIPTION
**Pull request by hzyi-google**
This PR allows setting up a field of the request object as an empty object using `{}`.

### CODE REVIEW

**Comment by hzyi-google** [comment]
PTAL

File: src/test/java/com/google/api/codegen/gapic/testdata/py/python_library.baseline [hunk 1]
```diff
@@ -6157,6 +6157,59 @@ def sample_get_book_from_absolutely_anywhere():
...
+
+def sample_get_shelf():
+  """Test setting up empty objects in the request objects."""
+
+  client = library_v1.LibraryServiceClient()
+
+  name = client.shelf_path('my-shelf')
+  options_ = ''
+
+  # we can set up this empty nested object now
```
Comment (beccasaurus) [inline src/test/java/com/google/api/codegen/gapic/testdata/py/python_library.baseline]: 🎉🎉🎉

File: src/test/java/com/google/api/codegen/protoannotations/testdata/csharp_library.baseline [hunk 2]
```diff
@@ -2451,6 +2451,61 @@ namespace Google.Example.Library.V1.Samples
...
+        /// Test setting up empty objects in the request objects.
+        /// </summary>
+        public static void SampleGetShelf()
+        {
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            GetShelfRequest request = new GetShelfRequest
+            {
+                ShelfName = new ShelfName("my-shelf"),
+                Options = "",
+                Message = new SomeMessage(),
```
Comment (beccasaurus) [inline src/test/java/com/google/api/codegen/protoannotations/testdata/csharp_library.baseline]: 🎉🎉🎉

File: src/test/java/com/google/api/codegen/gapic/testdata/ruby/ruby_library.baseline [hunk 3]
```diff
@@ -6557,6 +6557,56 @@ if $PROGRAM_NAME == __FILE__
...
+def sample_get_shelf
+  # Instantiate a client
+  library_client = Library::Library.new version: :v1
+
+  formatted_name = library_client.class.shelf_path("my-shelf")
+  options_ = ''
+
+  # we can set up this empty nested object now
+  message = {}
+  string_builder = {}
```
Comment (beccasaurus) [inline src/test/java/com/google/api/codegen/gapic/testdata/ruby/ruby_library.baseline]: 🎉🎉🎉

File: src/test/java/com/google/api/codegen/protoannotations/testdata/java_library.baseline [hunk 4]
```diff
@@ -3788,6 +3788,75 @@ public class TestResourceNameOneof2 {
...
+   */
+
+  /** Test setting up empty objects in the request objects. */
+  public static void sampleGetShelf() {
+    try (LibraryClient libraryClient = LibraryClient.create()) {
+      ShelfName name = ShelfName.of("my-shelf");
+      String options = "";
+
+      // we can set up this empty nested object now
+      SomeMessage message = SomeMessage.newBuilder().build();
```
Comment (beccasaurus) [inline src/test/java/com/google/api/codegen/protoannotations/testdata/java_library.baseline]: 🎉🎉🎉

File: src/test/java/com/google/api/codegen/protoannotations/testdata/nodejs_library.baseline [hunk 5]
```diff
@@ -1479,6 +1479,48 @@ function sampleGetBookFromAbsolutelyAnywhere() {
...
+
+/** Test setting up empty objects in the request objects. */
+function sampleGetShelf() {
+  const client = new library.LibraryServiceClient();
+  const formattedName = client.shelfPath('my-shelf');
+  const options = '';
+
+  // we can set up this empty nested object now
+  const message = {};
+  const stringBuilder = {};
```
Comment (beccasaurus) [inline src/test/java/com/google/api/codegen/protoannotations/testdata/nodejs_library.baseline]: 🎉🎉🎉

File: src/test/java/com/google/api/codegen/protoannotations/testdata/php_library.baseline [hunk 6]
```diff
@@ -1855,6 +1855,61 @@ function sampleGetBookFromAbsolutelyAnywhere()
...
+/** Test setting up empty objects in the request objects. */
+function sampleGetShelf()
+{
+    $libraryServiceClient = new LibraryServiceClient();
+
+    $formattedName = $libraryServiceClient->shelfName('my-shelf');
+    $options = '';
+
+    // we can set up this empty nested object now
+    $message = new SomeMessage();
```
Comment (beccasaurus) [inline src/test/java/com/google/api/codegen/protoannotations/testdata/php_library.baseline]: 🎉🎉🎉

File: src/test/java/com/google/api/codegen/protoannotations/testdata/python_library.baseline [hunk 7]
```diff
@@ -6233,6 +6233,59 @@ def sample_get_book_from_absolutely_anywhere():
...
+def sample_get_shelf():
+  """Test setting up empty objects in the request objects."""
+
+  client = library_v1.LibraryServiceClient()
+
+  name = client.shelf_path('my-shelf')
+  options_ = ''
+
+  # we can set up this empty nested object now
+  message = {}
```
Comment (beccasaurus) [inline src/test/java/com/google/api/codegen/protoannotations/testdata/python_library.baseline]: 🎉🎉🎉

**Comment by vchudnov-g** [comment]
LGTM. @hzyi-google , can you add a line for this to the internal feature spec spreadsheet?

**Pull request merged**